gpt_researcher/retrievers/pubmed_central/pubmed_central.py

```
@@ -1,174 +1,152 @@
         }
-        response = requests.get(base_url, params=params)
-
-        if response.status_code != 200:
-            raise Exception(
-                f"Failed to retrieve data: {response.status_code} - {response.text}"
-            )
-
-        return response.text
+        
+        try:
+            response = requests.get(self.base_search_url, params=search_params)
```

Ruff (0.14.3)

Probable use of `requests` call without timeout

(S113)